FirebaseAppCheck.podspec

```diff
@@ -75,20 +75,6 @@ Pod::Spec.new do |s|
     unit_tests.requires_app_host = true
   end
 
-  s.test_spec 'integration' do |integration_tests|
```

Comment:
Shouldn't integration tests still work? Presumably integration tests should be testing the public FirebaseAppCheck API?

Reply:
We could have an integration test that does something like:
```
AppCheck.setAppCheckProviderFactory(AppCheckDebugProvider())
FirebaseApp.configure()

AppCheck.appCheck().limitedUseToken { token, error in
  // Verify no error and valid token
}

AppCheck.appCheck().token(forcingRefresh: true) { token, error in
  // Verify no error and valid token
}
```
Specifically the debug provider so that it could run on CI. Potentially we could write the test in Swift. 

I deleted the existing integration test for now since it tests `FIRDeviceCheckAPIService` (not exposed publicly and I deleted the file in this PR). I'd prefer to tackle the new integration test in a different PR (after pruning `FIRAppCheck.m`) but could also be convinced to write it first. Any recommendations or preferences?

Reply:
OK, to move forward without it as long as it stays in the plan. Thanks!

Reply:
I brought back the integrations test target and added a placeholder `AppCheckE2ETests` with a TODO so it wouldn't get forgotten.